packages/medusa/src/api/routes/admin/inventory-items/delete-location-level.ts

```diff
@@ -0,0 +1,69 @@
+import { Request, Response } from "express"
+import { IInventoryService } from "../../../../interfaces"
+
+/**
+ * @oas [delete] /inventory-items/{id}/location-levels/{location_id}
+ * operationId: "DeleteInventoryItemsInventoryIteLocationLevelsLocation"
+ * summary: "Delete a location level of an Inventory Item."
+ * description: "Delete a location level of an Inventory Item."
+ * x-authenticated: true
+ * parameters:
+ *   - (path) id=* {string} The ID of the Inventory Item.
+ *   - (path) location_id=* {string} The ID of the location.
+ * x-codeSamples:
+ *   - lang: JavaScript
+ *     label: JS Client
+ *     source: |
+ *       import Medusa from "@medusajs/medusa-js"
+ *       const medusa = new Medusa({ baseUrl: MEDUSA_BACKEND_URL, maxRetries: 3 })
+ *       // must be previously logged in or use api token
+ *       medusa.admin.inventoryItems.deleteLocationLevel(inventoryItemId, locationId)
+ *       .then(({ inventory_item }) => {
+ *         console.log(inventory_item.id);
+ *       });
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request DELETE 'https://medusa-url.com/admin/inventory-items/{id}/location-levels/{location_id}' \
+ *       --header 'Authorization: Bearer {api_token}' \
+ *       --header 'Content-Type: application/json'
+ * security:
+ *   - api_token: []
+ *   - cookie_auth: []
+ * tags:
+ *   - Inventory Items
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           $ref: "#/components/schemas/AdminInventoryItemsRes"
+ *   "400":
+ *     $ref: "#/components/responses/400_error"
+ *   "401":
+ *     $ref: "#/components/responses/unauthorized"
+ *   "404":
+ *     $ref: "#/components/responses/not_found_error"
+ *   "409":
+ *     $ref: "#/components/responses/invalid_state_error"
+ *   "422":
+ *     $ref: "#/components/responses/invalid_request_error"
+ *   "500":
+ *     $ref: "#/components/responses/500_error"
+ */
+export default async (req: Request, res: Response) => {
+  const { id, location_id } = req.params
+
+  const inventoryService: IInventoryService =
+    req.scope.resolve("inventoryService")
+
+  await inventoryService.deleteInventoryLevel(id, location_id)
+
```

Comment:
thought: would it make sense to enforce that users must delete/update all reservations held at the location before the Inventory Level is deleted?

Reply:
I think we should allow users to force delete an inventory level. Since locations for reservations don't matter in this iteration I don't think it should be blocking. We probably should require that there still is enough stock at the sales channel before removing the location wdyt? 😄

Reply:
But in this case, we will have "orphaned" reservations. I.e. it doesn't make sense that a reservation on inventory item A exists at location X if location X doesn't hold inventory for item A. 

I think it is a fair precondition that no reservations can exist on an Inventory Level for it to be deletable.

Reply:
I added the validation of reserved items